tempesta_fw/ss_skb.h

```diff
@@ -71,39 +71,60 @@ ss_skb_queue_empty(const SsSkbList *list)
 static inline void
 ss_skb_queue_tail(SsSkbList *list, struct sk_buff *skb)
 {
-	SsSkbCb *scb = TFW_SKB_CB(skb);
-
 	/* The skb shouldn't be in any other queue. */
-	WARN_ON_ONCE(scb->next || scb->prev);
+	WARN_ON_ONCE(skb->next || skb->prev);
 
-	scb->prev = list->last;
+	skb->prev = list->last;
 	if (ss_skb_queue_empty(list))
 		list->first = skb;
 	else
-		TFW_SKB_CB(list->last)->next = skb;
+		list->last->next = skb;
 	list->last = skb;
 }
 
+/**
+ * Init @list to prepare for skbs from @skb->frag_list.
+ */
 static inline void
-ss_skb_unlink(SsSkbList *list, struct sk_buff *skb)
+ss_skb_init_from_frag_list(SsSkbList *list, struct sk_buff *skb)
+{
+	WARN_ON_ONCE(skb->next || skb->prev);
+
+	skb->next = skb_shinfo(skb)->frag_list;
+	list->first = list->last = skb;
+}
```

Comment:
Well, `skb_shinfo(skb)->frag_list` is just a pointer to `skb` - the proof that we don't need `SsSkbList` any more. Since the function will work just with plain `skb` pointers, just 2 lines in size and is used only once, there is no need to keep the function and the code can be directly mode to `ss_skb_unroll()`.

Reply:
Done.